ext/dom/dom.stub.php

```diff
@@ -0,0 +1,313 @@
+    public function removeAttribute(string $name) {}
+
+    /** @return DOMAttr|false */
+    public function removeAttributeNS(?string $namespaceURI, string $localName) {}
+
+    /** @return DOMAttr|false */
+    public function removeAttributeNode(string $name) {}
+
+    /** @return DOMAttr|bool */
+    public function setAttribute(string $name, string $value) {}
+
+    /** @return null|false */
+    public function setAttributeNS(?string $namespaceURI, string $localName, string $value) {}
+
+    /** @return DOMAttr|null|false */
+    public function setAttributeNode(DOMAttr $attr) {}
+
+    /** @return DOMAttr|null|false */
+    public function setAttributeNodeNS(DOMAttr $attr) {}
+
+    public function setIdAttribute(string $name, bool $isId) {}
+
+    public function setIdAttributeNS(string $namespaceURI, string $localName, bool $isId) {}
+
+    public function setIdAttributeNode(DOMAttr $attr, bool $isId) {}
+}
+
+class DOMEntityReference {
+    public function __construct(string $name) {}
+}
+
+class DOMImplementation {
+    public function getFeature(string $feature, string $version) {}
+
+    /** @return bool */
+    public function hasFeature(string $feature, string $version) {}
+
+    public function createDocumentType($qualifiedName, $publicId, $systemId) {}
+
+    /** @return DOMDocument|false */
+    public function createDocument(?string $namespaceURI = null, ?string $qualifiedName = null, ?DOMDocumentType $doctype = null) {}
+}
+
+class DOMNamedNodeMap {
+    /** @return ?DOMNode */
+    public function getNamedItem(string $name) {}
+
+    /** @return ?DOMNode */
+    public function getNamedItemNS(?string $namespaceURI, string $localName) {}
+
+    /** @return ?DOMNode */
+    public function item(int $index) {}
+
+    /** @return int|false */
+    public function count() {}
+}
+
+class DOMNodeList {
+    /** @return int|false */
+    public function count() {}
+
+    /** @return ?DOMNode */
+    public function item(int $index) {}
+}
+
+class DOMProcessingInstruction {
+    public function __construct(string $name, ?string $value = null) {}
+}
+
+class DOMText {
+    public function __construct(string $value) {}
+
+    /** @return bool */
+    public function isWhitespaceInElementContent() {}
+
+    /** @return bool */
+    public function isElementContentWhitespace() {}
+
+    /** @return DOMText|false */
+    public function splitText(int $offset) {}
+}
+
+class DOMXPath {
+    public function __construct(DOMDocument $doc) {}
+
+    public function evaluate(string $expr, ?DOMNode $context = null, bool $registerNodeNS = true) {}
+
+    public function query(string $expr, ?DOMNode $context = null, bool $registerNodeNS = true) {}
+
+    /** @return bool */
+    public function registerNamespace(string $prefix, string $namespaceURI) {}
+
+    /** @param string|array $restrict */
+    public function registerPhpFunctions($restrict = null) {}
+}
+
+function dom_import_simplexml(SimpleXMLElement $node): ?DOMElement {}
```

Comment:
Does this throw is a non-SimpleXMLElement is passed?

Reply:
No, it just emits a warning :-/ What is the approach in that case?

Reply:
I'd relay the type to `object` for now.